editor/src/components/editor/store/editor-state.ts

```diff
@@ -1259,11 +1259,11 @@ export function deriveState(
   )
 
   const derived: DerivedState = {
-    navigatorTargets: Utils.keepReferenceIfShallowEqual(
+    navigatorTargets: Utils.keepReferenceIfDeepEqualSLOW(
```

Comment:
Given the change to `Utils.keepReferenceIfDeepEqualSLOW` to now use `fast-deep-equal`, I think it probably makes sense to just switch these to use `keepDeepReferenceEqualityIfPossible`, since that is written based on `fast-deep-equal` anyway (assuming this switch was only because we were previously using `Utils.keepReferenceIfShallowEqual`)

Reply:
So I might revert the change to the `SLOW` suffixed function and switch over to that.

Reply:
Changes made in e52217f4.

Reply:
hmm, since the template paths are memoized, keepReferenceIfShallowEqual should be sufficient here. but I think there shouldn't be a significant difference because fast-deep-equals starts by doing a `===` check.